src/lib/communication/DY.Lib.Communication.RequestResponse.Lemmas.fst

```diff
@@ -243,28 +277,26 @@ let compute_response_message_proof #cinvs #a tr server key nonce request respons
   let ciphertext = aead_enc key nonce res_bytes ad_bytes in
   // Needed for the case that the key is publishable
   FStar.Classical.move_requires (aead_enc_preserves_publishability tr key nonce res_bytes) ad_bytes;
-  serialize_wf_lemma com_message_t (is_publishable tr) (ResponseMessage {nonce; ciphertext});
+  serialize_wf_lemma comm_message_t (is_publishable tr) (ResponseMessage {nonce; ciphertext});
   ()
 
 val send_response_proof:
   {|protocol_invariants|} ->
-  #a:Type -> {| parseable_serializeable bytes a |} ->
+  #a:eqtype -> {|comm_layer_reqres_config a|} ->
   tr:trace ->
   higher_layer_preds:comm_reqres_higher_layer_event_preds a ->
-  server:principal -> req_meta_data:comm_meta_data -> response:a ->
+  server:principal -> req_meta_data:comm_meta_data a -> response:a ->
   Lemma
   (requires
     trace_invariant tr /\
-    has_communication_layer_reqres_crypto_predicates /\
-    has_communication_layer_reqres_event_predicates request_response_event_preconditions higher_layer_preds /\
-    (match parse a req_meta_data.request with
-    | None -> False
-    | Some request -> higher_layer_preds.send_response tr server request response) /\
-    has_communication_layer_state_predicates /\
+    has_communication_layer_reqres_crypto_predicates a /\
+    has_communication_layer_reqres_event_predicates higher_layer_preds /\
+    higher_layer_preds.send_response tr server req_meta_data.request response /\
+    has_communication_layer_reqres_state_predicates a /\
```

Comment:
Here, for example, having one predicate that includes all of crypto/event/state at once would be cleaner.